feat: add new Token Permissions and Token Privacy Settings (#133)
Basis-Theory/docs: source/includes/api-reference/_tokens.md

| `prov`    | *string* | Optional encryption provider (e.g. AWS, AZURE, GCP, etc.) |
| `alg`     | *string* | Encryption algorithm (e.g. AES, RSA, etc)                 |

### Privacy Object

Token Privacy defines the privacy settings applied to a Token. By default, privacy settings will be applied based on the [Token Type](#tokens-token-types).
Default privacy settings can be overridden at the time of creation, but only to a setting with a higher specificity level.

| Attribute            | Type     | Description                                                                                                                                                                            |
|----------------------|----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `classification`     | *string* | [Classification](#tokens-token-classifications) of the Token (e.g. `general`, `bank`, `pci`)                                                                                           |

Reviewer: The two lines below use i.e., not e.g.
```suggestion
| `classification`     | *string* | [Classification](#tokens-token-classifications) of the Token (i.e. `general`, `bank`, `pci`)                                                                                           |
```

Author: `i.e.` is used when the list is exhaustive, but here we are only listing a partial set of the possible classifications supported.

|-------|---------|-----------------------------------------|------------------------|----------------------|----------------------|----------------------------|
| Token | `token` | Generic token type                      | `general`              | `high`               | `low`                | `redact`                   |
| Card  | `card`  | [Atomic card](#atomic-cards) token type | `pci`                  | `high`               | `high`               | `mask`                     |
| Bank  | `bank`  | [Atomic bank](#atomic-banks) token type | `bank`                 | `high`               | `high`               | `mask`                     |

## Token Classifications

Each token has a data classification associated with it which defines the type of data it contains.
Basis Theory permissions access to Tokens based on data classifications (see [Token Permissions](#permissions-token-permissions) for more details). 
The following data classifications are supported:

| Name      | Description | Specificity Level |
|-----------|-------------|-------------------|

> | `general` | TODO        | 0                 |

Author: ```suggestion
| `general` | Contains sensitive data that does not fall under a more specific classification or compliance requirements        | 0                 |
```

> | `bank`    | TODO        | 10                |

Author: ```suggestion
| `bank`    | Contains data that falls under banking compliance (e.g. Nacha)        | 10                |
```

> | `pci`     | TODO        | 10                |

Author: ```suggestion
| `pci`     | Contains data that falls under PCI compliance        | 10                |
```

> | `pii`     | TODO        | 10                |

Author: ```suggestion
| `pii`     | Contains user or customer specific identifiers (e.g. email, date of birth, address)        | 10                |
```

|------------|------------------------------------------------------------------------------------------------------------------------|-------------------|
| `low`      | Loss of data confidentiality, integrity, or availability is expected to have **limited** adverse effect                | 0                 |
| `moderate` | Loss of data confidentiality, integrity, or availability is expected to have **serious** adverse effect                | 1                 |
| `high`     | Loss of data confidentiality, integrity, or availability is expected to have **severe or catastrophic** adverse effect | 2                 |

## Token Restriction Policies

A Token Restriction Policy defines the policy to enforce on a Token's data when read by a User or Application
with permission to read the Token's classification but at a lower impact level.

| Name     | Description                                                                                                                 | Specificity Level |
|----------|-----------------------------------------------------------------------------------------------------------------------------|-------------------|

> | `mask`   | Token data will be masked if available, falling back to `redact` restriction policy if mask is undefined for the Token Type | 0                 |

Author: ```suggestion
| `mask`   | Token data will be masked in the response, falling back to `redact` restriction policy if a mask is not available for the Token Type | 0                 |
```

### Permissions

<p class="scopes">

>   <span class="scope">token:create (deprecated)</span>

Author: ```suggestion
```

  <span class="scope">token:&lt;classification&gt;:create</span>
</p>

### Request Parameters

| Attribute    | Required | Type                                                          | Default | Description                                                                                                                         |
|--------------|----------|---------------------------------------------------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------|
| `type`       | true     | *string*                                                      | `null`  | [Token type](#tokens-token-types) of the token                                                                                      |
| `data`       | true     | *any*                                                         | `null`  | Token data. Can be an object, array, or any primitive type such as an integer, boolean, or string                                   |

> | `privacy`    | false    | *[privacy object](#tokens-token-object-privacy-object)*       | `null`  | Token Privacy Settings overrides. Settings must be a higher specificity level than the default privacy settings for the Token type. |

Author: ```suggestion
| `privacy`    | false    | *[privacy object](#tokens-token-object-privacy-object)*       | `null`  | Token Privacy Settings overrides. Overrides must be a higher specificity level than the default or minimum setting for the [Token Type](#token-token-types). |
```

### Permissions

<p class="scopes">

>   <span class="scope">token:read (deprecated)</span>

Author: ```suggestion
```

### Permissions

<p class="scopes">

>   <span class="scope">token:decrypt (deprecated)</span>

Author: ```suggestion
  <span class="scope">token:&lt;classification&gt;:read:&lt;impact_level&gt;</span>
```

### Permissions

<p class="scopes">

>   <span class="scope">token:read (deprecated)</span>

Author: ```suggestion
```

  "type": "token",
  "tenant_id": "77cb0024-123e-41a8-8ff8-a3d5a0fa8a08",
  "data": "Sensitive Value",
  "privacy": {
    "classification": "general",
    "impact_level": "moderate",
    "restriction_policy": "redact"

>   },

Author: TODO, update permissions for decrypt token by ID

### Permissions

<p class="scopes">

>   <span class="scope">token:delete (deprecated)</span>

Author: ```suggestion
```
